src/imu/imu.cpp

```diff
@@ -0,0 +1,242 @@
+
+#include <yarp/os/Property.h>
+#include <yarp/os/Stamp.h>
+
+#include "imu.h"
+
+using namespace robottestingframework;
+ROBOTTESTINGFRAMEWORK_PREPARE_PLUGIN(Imu)
+
+Imu::Imu() : TestCase("Imu") { }
+
+Imu::~Imu() { }
+
+bool Imu::setup(yarp::os::Property& property) {
+    
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("robot"), "The robot name must be given as the test parameter!");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("port"), "The port name must be given as the test parameter!");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("model"), "Please, provide the urdf model path.");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("frame"), "Please, provide the frame name.");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("sensor"), "Please, provide the sensor name.");
+    
+    robotName = property.find("robot").asString(); // robot name
+    portName = property.find("port").asString(); // name of the port from which the data are streamed
+    modelPath = property.find("model").asString(); // urdf model path
+    frameName = property.find("frame").asString(); // frame on which the sensor is attached
+    sensorName = property.find("sensor").asString(); // sensor name within urdf
+
+    ROBOTTESTINGFRAMEWORK_TEST_REPORT("Running IMU test on "+robotName+"...");
+
+    yarp::os::Property options1;
+    options1.put("device", "multipleanalogsensorsclient");
+    options1.put("remote", portName);
+    options1.put("local", "/imuTest/"+portName);
+    
+    driver1 = new yarp::dev::PolyDriver(options1);
+    
+    yarp::os::Property options2;
+    options2.put("device", "remotecontrolboardremapper");
+    yarp::os::Bottle axesNames;
+    yarp::os::Bottle & axesList = axesNames.addList();
+    axesList.addString("neck_pitch");
+    axesList.addString("neck_roll");
+    axesList.addString("neck_yaw");
+    axesList.addString("torso_pitch");
+    axesList.addString("torso_roll");
+    axesList.addString("torso_yaw");
+    axesList.addString("l_shoulder_pitch");
+    axesList.addString("l_shoulder_roll");
+    axesList.addString("l_shoulder_yaw");
+    axesList.addString("l_elbow");
+    axesList.addString("r_shoulder_pitch");
+    axesList.addString("r_shoulder_roll");
+    axesList.addString("r_shoulder_yaw");
+    axesList.addString("r_elbow");
+
+    options2.put("axesNames", axesNames.get(0));
+
+    yarp::os::Bottle remoteControlBoards;
+    yarp::os::Bottle & remoteControlBoardsList = remoteControlBoards.addList();
+    remoteControlBoardsList.addString("/"+robotName+"/torso");
+    remoteControlBoardsList.addString("/"+robotName+"/head");
+    remoteControlBoardsList.addString("/"+robotName+"/left_arm");
+    remoteControlBoardsList.addString("/"+robotName+"/right_arm");
+    options2.put("remoteControlBoards", remoteControlBoards.get(0));
+    options2.put("localPortPrefix", "/test");
+
+    driver2 = new yarp::dev::PolyDriver(options2);
```

Comment:
I would change `options1`, `options2`, `driver1`, `driver2` to more meaningful name for ease the future code understanding

Reply:
What about `MASclientDriver/Options` and `controlBoardDriver/Options`?

Reply:
Ok for me!